pages/api/auth/[...nextauth].js

```diff
@@ -21,13 +22,29 @@ export const authOptions = {
     }),
   ],
   callbacks: {
+    async signIn({ user, account, profile, email, credentials }) {
+      const docSnap = await getDoc(doc(db, "users", user.id));
+      if (!docSnap.data().start) {
+        const currUser = doc(db, "users", user.id);
+
+        await updateDoc(currUser, {
+          username: user.email.split("@")[0],
+          role: "member",
+          start: Timestamp.now(),
+        });
+        return false;
```

Comment:
returning false here will actually cause the login to fail, you should just return true no matter what

Reply:
This is correct!